File: client/app/src/main/java/com/qubacy/geoqq/domain/mate/chats/model/MateChat.kt
package com.qubacy.geoqq.domain.mate.chats.model

import com.qubacy.geoqq.data.mate.chat.model.DataMateChat
import com.qubacy.geoqq.domain._common.model.user.User
import com.qubacy.geoqq.domain._common.model.user.toUser
import com.qubacy.geoqq.domain.mate.message.model.MateMessage
import com.qubacy.geoqq.domain.mate.message.model.toMateMessage

data class MateChat(
    val id: Long,
    val user: User,
    val newMessageCount: Int,
    val lastMessage: MateMessage?
) {

}

fun DataMateChat.toMateChat(): MateChat {
    return MateChat(id, user.toUser(), newMessageCount, lastMessage?.toMateMessage())
}

File: server/own_services/geoqq_ws/internal/adapters/interfaces/wsApi/internal/dto/serverSide/payload/mateChat.go
package payload

import (
	domain "common/pkg/domain/geoqq"
	utl "common/pkg/utility"
	"geoqq_ws/internal/constErrors"
)

/*

{
	"id": "<id>",
	"user-id": "<id>",
	"new-message-count": "<int>",
	"last-message": {
		"id": "<id>",
		"text": "<string>",
		"time": "<int>",
		"user-id": "<id>"
	},
	"last-action-time": "<int>"
}

*/

type MateChat struct {
	Id              float64      `json:"id"`
	UserId          float64      `json:"user-id"`
	NewMessageCount float64      `json:"new-message-count"`
	LastMessage     *MateMessage `json:"last-message,omitempty"`
	LastActionTime  float64      `json:"last-action-time"`
}

func MateChatFromDomain(dm *domain.MateChat) (*MateChat, error) {
	if dm == nil {
		return nil, constErrors.ErrInputParamWithTypeNotSpecified(
			utl.GetTypeName(dm))
	}

	return &MateChat{
		Id:              float64(dm.Id),
		UserId:          float64(dm.UserId),
		NewMessageCount: float64(dm.NewMessageCount),
		LastMessage:     MateMessageFromDomainOrNil(dm.LastMessage),
		LastActionTime:  float64(dm.LastActionTime.Unix()), // !
	}, nil
}
